sdk/core/azure-core/inc/http/request.hpp

```diff
@@ -10,138 +10,129 @@
     }
 
-    // Get query parameters string and update
-    auto queryParameters = url.substr(position + 1);
-    url = url.substr(0, position);
-
-    // Split all query parameters (should be separated by &)
-    auto queryParametersVector = split(queryParameters, '&');
-
-    // insert each query parameter to internal field
-    std::for_each(queryParametersVector.begin(), queryParametersVector.end(), [&](auto query) {
-      auto parameter = split(query, '=');
-      // Will throw if parameter in query is not valid (i.e. arg:1)
-      _queryParameters.insert(std::pair<std::string, std::string>(parameter[0], parameter[1]));
-    });
-
-    return url;
-  }
-
-  Request(
-      HttpMethod httpMethod,
-      std::string const& url,
-      BodyStream* bodyStream,
-      BodyBuffer* bodyBuffer)
-      : _method(std::move(httpMethod)), _url(parseUrl(std::move(url))), _bodyStream(bodyStream),
-        _bodyBuffer(bodyBuffer), _retryModeEnabled(false)
-  {
-    // TODO: parse url
-  }
+    /**
+     * Will check if there are any query parameter in url looking for symbol '?'
+     * If it is found, it will insert query parameters to _queryParameters internal field
+     * and remove it from url
+     */
+    std::string parseUrl(std::string url)
+    {
+      auto position = url.find('?');
 
-public:
-  Request(HttpMethod httpMethod, std::string const& url)
-      : Request(httpMethod, url, BodyStream::null, BodyBuffer::null)
-  {
-  }
+      if (position == std::string::npos)
+      {
+        return url; // no query parameters. Nothing else to do
+      }
 
-  Request(HttpMethod httpMethod, std::string const& url, BodyBuffer* bodyBuffer)
-      : Request(httpMethod, url, BodyStream::null, bodyBuffer)
-  {
-  }
+      // Get query parameters string and update
+      auto queryParameters = url.substr(position + 1);
+      url = url.substr(0, position);
 
-  Request(HttpMethod httpMethod, std::string const& url, BodyStream* bodyStream)
-      : Request(httpMethod, url, bodyStream, BodyBuffer::null)
-  {
-  }
-
-  // Methods used to build HTTP request
-  void addPath(std::string const& path);
-  void addQueryParameter(std::string const& name, std::string const& value);
-  void addHeader(std::string const& name, std::string const& value);
-  void startRetry(); // only called by retry policy
-
-  // Methods used by transport layer (and logger) to send request
-  HttpMethod getMethod();
-  std::string getEncodedUrl(); // should return encoded url
-  std::map<std::string, std::string> getHeaders();
-  BodyStream* getBodyStream();
-  BodyBuffer* getBodyBuffer();
-};
-
-} // namespace http
-} // namespace core
-} // namespace azure
+      // Split all query parameters (should be separated by &)
+      auto queryParametersVector = split(queryParameters, '&');
+
+      // insert each query parameter to internal field
+      std::for_each(queryParametersVector.begin(), queryParametersVector.end(), [&](auto query) {
+        auto parameter = split(query, '=');
```

Comment:
I remember `=` in query parameter value part can be encoded but not necessary according to specification. That means query parameter like `?key1=value1=1&key2=value2` is valid. So I don't think it's okay to use `split` here.